### PR TITLE
withAuthenticator docs from readme

### PR DESCRIPTION
**Pull request by jordanranz**
*Description of changes:*
- `withAuthenticator` docs
- Use JSX for code blocks

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/ui/auth/fragments/react/authenticator.md [hunk 1]
```diff
@@ -326,3 +351,27 @@ const App = () => (
...
+
+```jsx
+export default withAuthenticator(App);
+```
+
+### Breaking changes for withAuthentciator
+
+<amplify-callout warning>
+
+If you were providing additional options to `withAuthenticator` (e.g. `includeGreetings`, `authenticatorComponents`, `federated`, `theme`), these have changed
```
Comment (brene) [inline docs/ui/auth/fragments/react/authenticator.md:369]: add a period to the sentence.

Comment (brene) [reply docs/ui/auth/fragments/react/authenticator.md:369]: and provide a sentence maybe like: "Review the updated properties [here](#properties)"

Comment (jordanranz) [reply docs/ui/auth/fragments/react/authenticator.md]: done

File: docs/ui/auth/fragments/react/authenticator.md [hunk 2]
```diff
@@ -278,6 +278,31 @@ const App = () => (
...
 <ui-component-props tag="amplify-greetings"></ui-component-props>
 
+### withAuthenticator
+
+The `withAuthenticator` is a higher-order component (HoC) that wraps `AmplifyAuthenticator`.
+
+**Usage**
+```jsx
+import React from 'react';
+import { withAuthenticator, AmplifySignOut} from '@aws-amplify/ui-react';
```
Comment (sammartinez) [inline docs/ui/auth/fragments/react/authenticator.md:288]:
```suggestion
import { withAuthenticator, AmplifySignOut } from '@aws-amplify/ui-react';
```

**Pull request merged**